File: swift/15.swift
func pascalTriangle(_ n: Int) -> [[Int]] {
  let needToGenerate = 2 * n - 3
  var triangle = [[1, 1]]

  for i in 0...needToGenerate {
    var triangleRow = [1]
    for j in 1..<triangle[i].count {
      triangleRow.append(triangle[i][j - 1] + triangle[i][j])
    }
    triangleRow.append(1)
    triangle.append(triangleRow)
  }

  return triangle
}

let num = 20
let triangle = pascalTriangle(num)
var sum = 0

var i = num - 2
var j = 0
while i < triangle.count && j <= num {
  sum += triangle[i][j]

  i += 1
  j += 1
}

print(sum)
// 137_846_528_820

File: julia/15.jl
function pascalTriangle(n)
	needToGenerate = 2n-3
	triangle = [[1, 1]]

	for i = 1:needToGenerate
		triangleRow = [1]
		for j = 2:length(triangle[i])
			push!(triangleRow, triangle[i][j-1] + triangle[i][j])
		end

		push!(triangleRow, 1)
		push!(triangle, triangleRow)
	end

	return triangle
end

num = 21 # Julia is 1-indexed (for some reason)
triangle = pascalTriangle(num)
sum = 0

i = num-2
j = 1

while i < length(triangle) && j <= num
	global sum += triangle[i][j]

	global i += 1
	global j += 1
end

println(sum)
# 137846528820
